src/main/java/frc/robot/subsystems/swerve/factories/swerveconstants/SwerveConstantsFactory.java

```diff
@@ -2,13 +2,20 @@
 
 import frc.robot.Robot;
 import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.SwerveName;
 
 public class SwerveConstantsFactory {
 
-    public static SwerveConstants create(){
+    public static SwerveConstants create(SwerveName swerveName) {
+        return switch (swerveName) {
+            case SWERVE -> createSwerveConstants();
+        };
+    }
+
+    private static SwerveConstants createSwerveConstants() {
         return switch (Robot.ROBOT_TYPE) {
-            case REAL, REPLAY -> RealSwerveConstants.swerveConstants;
-            case SIMULATION -> SimulationSwerveConstants.swerveConstants;
+            case REAL, REPLAY -> RealSwerveConstants.getSwerveConstants(SwerveName.SWERVE);
+            case SIMULATION -> SimulationSwerveConstants.getSwerveConstants(SwerveName.SWERVE);
```

Comment:
do injection to the swerve name

Reply:
but this is supposed to be the function for SwerveName.SWERVE only, it will be that name either way

Reply:
Yea I know but like that the getSwerveconstants should not use injection either, you can find better solution instead of injection or use injection , but with injection it’s one less place to change when changing

Reply:
but you shouldn't be changing this. these are constants for SWERVE1 and you don't want to create a swerve with SWERVE1 constants and SWERVE2 SwerveName

Reply:
you can not change if you don't want, you can resolve